Fix minor markdown issues found

kmcgrady opened the pull request:
* Padding of Ordered Lists
* Color of HRs
* Margin/Padding of list items

Commit:
Fix Markdown issues found

akrolsmir reviewed (approved):
On PR Preview:

- I can see the improvement on lists (ordered and unordered)!
- ~Can't find the difference in `<hr>`. but also --where do we even use hr? Am I correct that `<hr>`s only show up through `st.components.v1.html(<hr></hr>)`~
Actually, looks like `<hr>` still isn't darker?
- Doesn't seem like the checkbox padding thing is handled in this PR (which is fine!)

```
import streamlit as st

st.markdown("""
Testing

1. Hello
2. Goodbye

---

- unordered
- list
""")

st.components.v1.html("<hr></hr>")

st.checkbox('yolo swag')
```

akrolsmir reviewed (changes requested):
Actually -- want to check in on the `<hr>` thing; my eyes can't tell a difference between currentColor and transparent here (or maybe it's just my monitor?)

Before:
![](https://i.imgur.com/3Hntbtz.png)

After:
![](https://i.imgur.com/gj7WE1i.png)

kmcgrady commented:
Re `<hr>`: `currentcolor` refers to the color of the text in that space. Means it's dependent on color, so it's possible the correct color was specified. Might also depend on browsers as well. Also, it's intentional to be lighter (I might disagree with the choice, but it's more about maintaining the original look).

Re Checkbox: It's a baseweb update, and since I updated the BaseWeb PR, I added the fix in there.

akrolsmir reviewed (approved):
Ah, makes sense. I checked out an older PR preview and it does seem like `<hr>` always used to inherit; not sure how Thiago got that screenshot, and whether `<hr>` should be darker can be another discussion.

Thanks for the clarification!
